### PR TITLE
Issue 16385 dot select

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: projects/dotcms-field-elements/src/components/dot-select/dot-select.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    beforeEach(async () => {
+        page = await newE2EPage();
+
+        await page.setContent(`
+        <dot-select
+            name="testName"
+            label="testLabel"
+            hint="testHint"
+            options="|,valueA|1,valueB|2"
```
Comment (fmontes) [inline projects/dotcms-field-elements/src/components/dot-select/dot-select.e2e.ts]: If we have spaces in the values or special chars did it work?

File: projects/dotcms-field-elements/src/components/dot-select/dot-select.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.dot-field__select--error {
+    border: 1px solid red;
```
Comment (fmontes) [inline projects/dotcms-field-elements/src/components/dot-select/dot-select.scss]: Are we doing this? I don't think is a good idea to add colors and/or any styles to the error state of the field.

File: projects/dotcms-field-elements/src/components/dot-select/readme.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# dot-dropdown
```
Comment (fmontes) [inline projects/dotcms-field-elements/src/components/dot-select/readme.md]: Shouldn't say here `dot-select`?

File: projects/dotcms-field-elements/src/components/dot-select/dot-select.e2e.ts (hunk 1 above)
Comment (alfredo-dotcms) [reply projects/dotcms-field-elements/src/components/dot-select/dot-select.e2e.ts]: yes

File: projects/dotcms-field-elements/src/components/dot-select/dot-select.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        this._dotTouched = true;
+        this.value = event.target.value;
+        this.emitValueChange();
+        this.emitStatusChange();
+    }
+
+    private emitInitialValue() {
+        if (!this.value) {
+            this.value = this._options[0].value;
+            this.emitValueChange();
```
Comment (freddyucv) [inline projects/dotcms-field-elements/src/components/dot-select/dot-select.tsx]: I thin this is the only one componente who emit the initial value, if we are doing this here we have to this in the others components

File: projects/dotcms-field-elements/src/components/dot-textarea/dot-textarea.tsx [hunk 5]
```diff
@@ -65,9 +65,10 @@ export class DotTextareaComponent {
     render() {
         return (
             <Fragment>
-                <label>{this.label}</label>
+                <label htmlFor={this.name}>{this.label}</label>
                 <textarea
                     class={this.getInputClassName()}
+                    id={this.name}
```
Comment (freddyucv) [inline projects/dotcms-field-elements/src/components/dot-textarea/dot-textarea.tsx]: the field name is not unique should be the variable

File: projects/dotcms-field-elements/src/components/dot-select/dot-select.tsx (hunk 4 above)
Comment (alfredo-dotcms) [reply projects/dotcms-field-elements/src/components/dot-select/dot-select.tsx]: no, this component is unique and when you reset it, it needs to behave differently from others, so you need to pick the first option of the dropdown and then emit it.

on the others `elements` the value is being set here https://github.com/dotCMS/core-web/blob/2d20298faddd03d29f2d1087cc8716c96c0369b3/projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx#L72

File: projects/dotcms-field-elements/src/components/dot-textarea/dot-textarea.tsx (hunk 5 above)
Comment (alfredo-dotcms) [reply projects/dotcms-field-elements/src/components/dot-textarea/dot-textarea.tsx]: image

we are currently using `variable` param

**Pull request merged**